docs/tutorials/getting-started-with-csharp/working-with-linq.md

```diff
@@ -0,0 +1,510 @@
+# Introduction
+This tutorial teaches you a number of features in .NET Core and the C# language. You’ll learn:
+
+*	How to generate sequences with LINQ
+*   How to write methods that can be easily used in LINQ queries.
+*   How to distinguish between eager and lazy evaluation.s 
+
+You'll learn these techniques by building an application that demonstrates
+one of the basic skills of any magician: the
+[faro shuffle](https://en.wikipedia.org/wiki/Faro_shuffle). Briefly,
+a faro shuffle is a technique where you split a card deck exactly in half,
+then the shuffle interleaves each one card from each half to rebuild the
+original deck.
+
+Magicians use this technique because every card is in a known location
+after each shuffle, and the order is a repeating pattern. 
+
+For our purposes, it is a light hearted look at manipulating sequences
+of data. The application you'll build will construct a card deck, and
+then perform a sequence of shuffles, writing the sequence out each time.
+You'll also compare the updated order to the original order.
+
+This tutorial has multiple steps. After each step, you can run the
+application and see the progress.
+
+
+# Prerequisites
+You’ll need to setup your machine to run .NET core. Instructions are
+[here](http://dotnet.github.io/getting-started/). You can run this
+application on Windows, Ubuntu Linux, OS X or in a Docker container. 
+You’ll need to install your favorite code editor. The descriptions below
+use [Visual Studio Code](https://code.visualstudio.com/) which is an open
+source, cross platform editor. However, you can use whatever tools you are
+comfortable with.
+
+# Create the Application
+
+The first step is to create a new application. Open a command prompt and
+create a new directory for your application. Make that the current
+directory. Type the command "dotnet new" at the command prompt. This
+creates the starter files for a basic “Hello World” application.
+
+If you've never used C# before, [this tutorial](console-teleprompter.md)
+explains the structure of a C# program. You can read that and then
+return here to learn more about LINQ. 
+
+# Creating the Data Set
+
+Let's start by creating a deck of cards. You'll do this using a LINQ
+query that has two sources (one for the four suites, one for the
+thirteen values). You'll combine those source into a 52 card deck.
+
+Here's the query:
+
+```cs
+var startingDeck = from s in Suits()
+                   from r in Ranks()
+                   select new { Suit = s, Rank = r };
+```
+
+The multiple `from` clauses produces a `SelectMany`, which creates
+a single sequence from combining each element in the first sequence
+with each element in the second sequence. The order is important
+for our purposes. The first element in the first source sequence
+(Suits) is combined with every element in the second sequence (Values).
+This produces the full sequence of first suite. That process is repeated
```

Comment:
This sentence on line 66 is unclear to me - you mean "suit". Maybe clarify to "This produces the full sequence of every card in the first suit." - if that's what you mean.